engine/src/test/java/org/hibernate/search/testsupport/junit/SearchFactoryHolder.java

```diff
@@ -152,4 +154,14 @@ public BatchBackend getBatchBackend() {
 		return batchBackend;
 	}
 
+	/**
+	 * The default for the tests in hibernate-search-engine is to not rely on JPA
+	 * specific annotations. These are typically used though when integrating
+	 * with Hibernate, so some tests might want to explicitly enable this.
+	 */
+	public SearchFactoryHolder enableJPAAnnotationsProcessing(boolean enableJPAAnnotationsProcessing) {
```

Comment:
Not related to this PR, but it seems we're introducing two different naming schemes in this class for option setters:

 * `withMultitenancyEnabled`, `withIdProvidedImplicit`, ...
 * `multipleInstances`, `enableJPAAnnotationsProcessing`, ...

No need to change this PR, but just so I know, do you have a preference?